pkg/pillar/cmd/zedmanager/memorysizemgmt.go

```diff
@@ -11,6 +11,30 @@ import (
 	"github.com/lf-edge/eve/pkg/pillar/utils/persist"
 )
 
+// getMemoryReservedForEveInBytes returns the amount of memory reserved for eve
+// in bytes. There are two sources for this value:
+// 1. Global config value `EveMemoryLimitInBytes`
+// 2. Global config value `EveMemoryLimitInMiB`
+// The first onve is the legacy config, as it does not support values more than 4GB.
+// But we still support it for backward compatibility. If it's set to valid value,
+// we use it. If it's set to 0 (which means the value set might be too high),
+// we fallback to the second one.
```

Comment:
On all existing devices I think we will have a default memory.eve.limit.bytes (persisted in /persist/status/zedagent/ConfigItemValueMap/global.json) thus the question is what steps does the user need to take to use a larger limit? Set memory.eve.limit.bytes to zero and set the new memory.eve.limit.MiB?

This, plus the existence of memory.eve.limit.MiB, needs to be documented in CONFIG-PROPERTIES.md

Reply:
> On all existing devices I think we will have a default memory.eve.limit.bytes (persisted in /persist/status/zedagent/ConfigItemValueMap/global.json) thus the question is what steps does the user need to take to use a larger limit? Set memory.eve.limit.bytes to zero and set the new memory.eve.limit.MiB?

Yes, we prepopulate the `memory.eve.limit.bytes` value based on the EVE cgroup limit at device startup, which is configured via the `dom0_mem` kernel argument. To apply a new memory limit, the user should update the `dom0_mem` kernel argument in the GRUB configuration to specify the desired cgroup limit. Upon the next EVE boot, the new memory limit will be stored in both `memory.eve.limit.bytes` and `memory.eve.limit.MiB`.

If the specified value fits within the `*.bytes` configuration, `memory.eve.limit.bytes` will be used by default.
If the byte value exceeds the allowable limit, it will be set to 0, and the system will then use the value from `memory.eve.limit.MiB` instead.



> This, plus the existence of memory.eve.limit.MiB, needs to be documented in CONFIG-PROPERTIES.md

In addition to `memory.eve.limit.MiB`, we should also document `memory.eve.limit.bytes`. However, we need to carefully consider whether to expose these options to users at this time. If users can modify this value, it would only affect the Pillar code without altering the actual cgroup limit, leading to potential inconsistencies.

We are planning to make memory limits editable through configuration settings in the future, allowing users to adjust both the config options and the cgroup limits seamlessly. Until this feature is implemented, I recommend not exposing `memory.eve.limit.*` in the documentation.

Alternatively, if we decide to include it, we should clearly advise users against manually setting these values, as it is automatically populated at EVE startup based on the corresponding kernel arguments.

Reply:
> Yes, we prepopulate the `memory.eve.limit.bytes` value based on the EVE cgroup limit at device startup, which is configured via the `dom0_mem` kernel argument. To apply a new memory limit, the user should update the `dom0_mem` kernel argument in the GRUB configuration to specify the desired cgroup limit. Upon the next EVE boot, the new memory limit will be stored in both `memory.eve.limit.bytes` and `memory.eve.limit.MiB`.

Now I'm confused (not by your PR but by the implications of how the current code works).
Can the user not set memory.eve.limit.bytes and have that apply (immediately or after a reboot)? It seems odd to have this memory.eve.limit.bytes config items if the value is taken from the dom0_mem kernel boot argument. Then what is the point of the memory.eve.limit.bytes?
(And that question is separate from adding the Mib config item.)

Reply:
I also find the current behavior of `memory.eve.limit.bytes` confusing =)
Originally, I guess, the intention was to allow users to set this option, which would affect both the cgroup memory limits and the Pillar code. However, this was never fully implemented. In its current state, setting `memory.eve.limit.bytes` does not actually modify the cgroup memory limits, and the value is overwritten on each device reboot based on the `dom0_mem` kernel argument. Additionally, there is inconsistency in the code where some parts read the memory limit from the configuration while others read it directly from the cgroups limits file, such as in [handlemetrics.go](https://github.com/lf-edge/eve/blob/5b2acf4b2be581e51b67592a273f77f8711b2901/pkg/pillar/cmd/zedagent/handlemetrics.go#L625). This results in two separate sources for the same value, leading to potential inconsistencies if a user manually sets `memory.eve.limit.bytes` without adjusting the corresponding kernel arguments. Currently, the `memory.eve.limit.bytes` configuration serves primarily as a global storage placeholder rather than an active configuration parameter. However, having this configuration in place is beneficial as it simplifies the future implementation of user-defined memory limit settings, where users will be able to adjust both the configuration options and the cgroup limits seamlessly.

Reply:
Then I agree we shouldn't document these in CONFIG-PROPERTIES.md.
But it would be good to create an issue to clean this up to not have two sources of config and confusion.

Reply:
> But it would be good to create an issue to clean this up to not have two sources of config and confusion.

Create, EV-1329